aten/src/ATen/cpu/vec/vec256/vec256_int.h

```diff
@@ -496,34 +496,34 @@ class Vectorized<int16_t> : public Vectorizedi {
   Vectorized<int16_t> le(const Vectorized<int16_t>& other) const;
 };
 
-template <>
-class Vectorized<int8_t> : public Vectorizedi {
+template <typename T>
+class Vectorized<T, std::enable_if_t<std::is_same<T, int8_t>::value || std::is_same<T, uint8_t>::value>> : public Vectorizedi {
```

Comment:
I'm not a fan of having all these `if (is_same_v<T, int8_t>)` branches everywhere. What was wrong with the common base class idea?

Reply:
IMO this is an approach towards maximal generalization, there is a spectrum of other approaches between previous one and this one.  I pushed this as an example (apparently, CI is showing there are compile errors in AVX-512 code that I'll have to fix, but 256-bit vectorization seems to be fine), what you mentioned is another valid approach.  We're talking about personal tastes at the moment, maybe as @lezcano suggested the previous version should be merged, and this refactoring postponed for later.

Reply:
IMO, it may be best to merge this PR as it was and then do this work in a follow-up PR.
FWIW, I agree with @peterbell10 that having a common base-class and implementing this via CRTP may be the most standard way to implement this. But again, we can do this in a follow-up PR.

Reply:
OK, the PR is ready for review then.  My personal preference is to see these small differences in the implementation next to each other, but I get other points of view.  I can keep changing it, but it takes time to go through CI cycles over and over again, and I have another small PR that depends on this - so I'd too prefer to merge as is.  Whenever this eventually refactored further, it should be noted that there is some commonality with the implementation of other datatypes, so that may be also taken into account.

Reply:
This PR was ready to go, pretty much approved, and passing CI in https://github.com/pytorch/pytorch/commit/06ee3b7e682dc99f615b53a84801305e3fa60cf5. 
Can't we revert to that state and get it merged? 
You can even split the diff between that commit and the current commit into a different PR, and we can discuss further changes in that PR.

Reply:
Done, the PR containing int8/uint8 refactoring is #89650, let's continue the discussion there.